UnityAds/UnityAdsInterstitialCustomEvent.m

```diff
@@ -126,84 +125,70 @@ - (BOOL)enableAutomaticImpressionAndClickTracking
+  [self.delegate fullscreenAdAdapterWillLeaveApplication:self];
 }
 
-- (void) unityAdsDidFinish:(NSString *)placementId withFinishState:(UnityAdsFinishState)state
-{
-    [UnityAds removeDelegate:self];
-    [self.delegate fullscreenAdAdapterAdWillDisappear:self];
-    MPLogAdEvent([MPLogEvent adWillDisappearForAdapter:NSStringFromClass(self.class)], [self getAdNetworkId]);
+- (void)unityAdsShowComplete:(NSString *)placementId withFinishState:(UnityAdsShowCompletionState)state {
+  [self.delegate fullscreenAdAdapterAdWillDisappear:self];
+  MPLogAdEvent([MPLogEvent adWillDisappearForAdapter:NSStringFromClass(self.class)], [self getAdNetworkId]);
 
-    [self.delegate fullscreenAdAdapterAdDidDisappear:self];
-    MPLogAdEvent([MPLogEvent adDidDisappearForAdapter:NSStringFromClass(self.class)], [self getAdNetworkId]);
-    
-    // Signal that the fullscreen ad is closing and the state should be reset.
-    // `fullscreenAdAdapterAdDidDismiss:` was introduced in MoPub SDK 5.15.0.
-    if ([self.delegate respondsToSelector:@selector(fullscreenAdAdapterAdDidDismiss:)]) {
-        [self.delegate fullscreenAdAdapterAdDidDismiss:self];
-    }
-}
+  [self.delegate fullscreenAdAdapterAdDidDisappear:self];
+  MPLogAdEvent([MPLogEvent adDidDisappearForAdapter:NSStringFromClass(self.class)], [self getAdNetworkId]);
 
-- (void) unityAdsDidClick:(NSString *)placementId
-{
-    MPLogAdEvent([MPLogEvent adTappedForAdapter:NSStringFromClass(self.class)], [self getAdNetworkId]);
-    [self.delegate fullscreenAdAdapterDidReceiveTap:self];
-    [self.delegate fullscreenAdAdapterDidTrackClick:self];
-    MPLogAdEvent([MPLogEvent adWillLeaveApplicationForAdapter:NSStringFromClass(self.class)], [self getAdNetworkId]);
-    [self.delegate fullscreenAdAdapterWillLeaveApplication:self];
+  // Signal that the fullscreen ad is closing and the state should be reset.
+  // `fullscreenAdAdapterAdDidDismiss:` was introduced in MoPub SDK 5.15.0.
+  if ([self.delegate respondsToSelector:@selector(fullscreenAdAdapterAdDidDismiss:)]) {
+      [self.delegate fullscreenAdAdapterAdDidDismiss:self];
+  }
 }
 
-- (void)unityAdsPlacementStateChanged:(nonnull NSString *)placementId oldState:(UnityAdsPlacementState)oldState newState:(UnityAdsPlacementState)newState {
-    // no-op
-}
+- (void)unityAdsShowFailed:(NSString *)placementId withError:(UnityAdsShowError)error withMessage:(NSString *)message {
 
-- (void)unityAdsAdLoaded:(NSString *)placementId {
-    [self.delegate fullscreenAdAdapterDidLoadAd:self];
-    MPLogAdEvent([MPLogEvent adLoadSuccessForAdapter:NSStringFromClass(self.class)], [self getAdNetworkId]);
+  if (error == kUnityAdsErrorShowError) {
+    NSError *showError= [self createErrorWith:@"Unity Ads failed to show interstitial"
+                                    andReason:message
+                                andSuggestion:@""];
+      MPLogAdEvent([MPLogEvent adShowFailedForAdapter:NSStringFromClass(self.class) error:showError], [self getAdNetworkId]);
```

Comment:
Also, now that we have placementId within show error, we can provide that information through their logs

Reply:
Yeah, I was on the fence about whether I should change log messages, but I guess just adding the placement ID to it can't hurt things too much.  I'll add it in and see if they have issues with it.